### PR TITLE
Pin knative/pkg and knative/test-infra to release-0.17

### PR DESCRIPTION
**Pull request by Harwayne**
## Proposed Changes

- Pin knative/pkg and knative/test-infra to `release-0.17`.
    - In preparation for `release-0.17` of knative/eventing in a week.

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: test/e2e-wi-tests.sh [hunk 1]
```diff
@@ -157,7 +157,7 @@ function gcp_auth_setup() {
 
 # Create a cluster with Workload Identity enabled.
 # We could specify --cluster-version to force the cluster using a particular GKE version.
-initialize $@ --cluster-creation-flag "--workload-pool=\${PROJECT}.svc.id.goog"
+initialize $@ --kubetest2-flag "--workload-pool=\${PROJECT}.svc.id.goog"
```
Comment (chizhg) [inline test/e2e-wi-tests.sh:160]: This should now be
```suggestion
initialize $@ --kubetest2-flag "--enable-workload-identity=true"
```

Comment (Harwayne) [reply test/e2e-wi-tests.sh]: Done. And thank you for the fix upstream!

**Pull request merged**